### DISCUSSION TITLE
Any component names?

### DISCUSSION THREAD

**Discussion by aolko** [realmorg/realm#62] [Q&A]
Can you amend [the stupid w3c rule](https://html.spec.whatwg.org/multipage/custom-elements.html#valid-custom-element-name), allowing "non-exclusively-dashed-component-names" [like riot does](https://riot.js.org/faq/#should-i-use-a-dash---in-the-tag-name)?

**Comment by oknoorap** [comment]
@aolko, We don't support that, because Realm uses `customElements.define` under the hood and doesn't rely on a bundler/compiler or the Node.js ecosystem. It solely depends on the JavaScript runtime in the browser.

**Reply by aolko** [reply]
Maybe write a custom implementation of custom elements **that does support any names**?

**Reply by oknoorap** [reply]
@aolko, I don't have any plans right now because Realm uses a straightforward implementation of customElements.define. However, I will consider it for the next version. 🙂

**Reply by aolko** [reply]
do you have a roadmap or ETA for that?

**Reply by oknoorap** [reply]
@aolko, I don't know. Maybe v2 will be released next year. I plan to rewrite Realm's core with better approach. I also wan to publish this framework known as Hypership (the holy trinity: HTMX, Alpine.js, and Tailwind). 🙂 

https://twitter.com/oknoorap/status/1746201757997486305

**Reply by aolko** [reply]
[master](https://rc.css.master.co/) > [unocss](https://unocss.dev/) > tailwind
& [alpine-ajax](https://alpine-ajax.js.org/) ~ htmx

**Comment by aolko** [comment]
See also: 
- https://github.com/11ty/webc
- https://vuejs.org/guide/scaling-up/sfc.html

**Reply by oknoorap** [reply]
In Realm, you can use whatever name you want with the `text/html` MIME type, but the custom element's name still needs to follow the W3C rules.

This is actually what sets realm apart from others; we don't need any compiler. It just works like this.

```html
<!-- https://your-domain.tld/anyname.html -->
<!-- please enable CORS -->

<custom-element name="counter-element">
  <element-state name="count" type="number">0</element-state>
  <element-flow>
    <trigger-event click="AddCountButton">
      <set-state name="count" value="1" mutate="+"></set-state>
    </trigger-event>
  </element-flow>

  <template>
    <button ref="AddCountButton">
      Add count <slot name="#count"></slot>
    </button>
  </template>
</custom-element>

<!-- In your codes from https://another.tld -->
<import-element from="https://your-domain.tld/anyname.html"></import-element>
<counter-element></counter-element>
```

**Reply by aolko** [reply]
the whole point is breaking https://html.spec.whatwg.org/multipage/custom-elements.html#valid-custom-element-name and not caring about W3C

**Reply by oknoorap** [reply]
@aolko, yeah, if we're using a JS runtime on a browser, I think we need to follow the W3C rules. And yes, it's possible if we have a compiler. Unfortunately, we don't currently do that. Even if we were using a compiler, the output should still adhere to the W3C rules. 😀

But thanks for providing me with more references to other libraries/frameworks (e.g Riot and Vue).

**Comment by Lukeduke14148686** [comment]
Please tell me what I did when I opened these files I'm just learning ill attach screenshots any help is appreciated
![Resized_Screenshot_20240108_180159_Chrome_152181479576146](https://github.com/realmorg/realm/assets/158804560/4f8b1bfa-f4c4-484e-9c49-79db5c906502)
![Screenshot_20240213_124506_HTML Viewer](https://github.com/realmorg/realm/assets/158804560/78a2491b-2451-46dd-bb4e-2b034089d949)
Please help me and help me understand ty

**Reply by oknoorap** [reply]
is this scam 😮
